omod/src/main/java/org/openmrs/module/datafilter/extension/html/Content.java

```diff
@@ -49,15 +51,17 @@ public String generate() {
 	private String addHTMLForLocationNames() {
 		return locationNames.stream().reduce("", (appendedContent, locationName) -> {
 			String checkedValue = "";
-			appendedContent += "<span class='listItem'>";
 			if (selectedLocations.contains(locationName)) {
 				checkedValue = CHECKED;
 			}
-			appendedContent += "<input type='checkbox' name='locationStrings' id='locationStrings." + locationName + "'"
-			        + " value='" + locationName + "'" + checkedValue + ">";
-			appendedContent += "<label for='locationStrings." + locationName + "'>" + locationName + "</label>";
+			locationName = StringEscapeUtils.escapeHtml(locationName);
```

Comment:
Can you try this:
```java
import org.springframework.web.util.HtmlUtils;
...
locationName = HtmlUtils.htmlEscape(locationName);
```
?
This should free you from requiring all the double quotes to single quotes changes, while actually working when form-submitting the values.

Of course you'll have to double check at app runtime but I believe that this will work as I tried a minimal example on W3C. For instance you can try this snippet containing the escaped HTML `value='Location&#39;1'` [here](https://www.w3schools.com/tags/tryit.asp?filename=tryhtml5_input_type_checkbox):
```html
<!DOCTYPE html>
<html>
<body>

<h3>Submitting a location name with an apostrophe</h3>

<form action="/action_page.php">

  <input type='checkbox' name='loc1' value='Location&#39;1'>
  <label for='loc1'>Location'1</label><br>

  <input type="submit" value="Submit">

</form>

</body>
</html>

```

Reply:
Done